Add documentation for slab benchmark multiphysics problem. (#99)
idaholab/salamander: doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md

# 1D Slab Benchmark: Neutron Transport Coupled With Thermal Conduction and Material Expansion

> This verification problem presents capabilities to couple neutron transport with thermal conduction and material expansion. The analytic benchmark slab problem [!citep](griesheimer2022analytical) describes a slab where S$_2$ neutron transport is coupled with thermal conduction, convective boundary conditions, Doppler-broadened cross sections, and nonlinear thermal expansion effects along the heated slab. The benchmark provides the analytic solutions for steady-state temperature, neutron flux, heating, multiplication factor, and the heated slab length.

Reviewer: ```suggestion
This verification problem demonstrates the ability to couple neutron transport with thermal conduction and material expansion. The analytic benchmark slab problem [!citep](griesheimer2022analytical) describes a slab where S$_2$ neutron transport is coupled with thermal conduction, convective boundary conditions, Doppler-broadened cross sections, and nonlinear thermal expansion effects along the heated slab. The verification provides the analytic solutions for steady-state temperature, neutron flux, heating, multiplication factor, and the heated slab length.
```

While the restriction to S$_2$ neutron transport would be considered overly simplistic for most radiation transport  simulations, the objective here is to verify thermomechanical-OpenMC coupling using analytic solutions for three coupled solution fields (displacements, temperature, and flux). Simplifications of each individual physics included in this model are not imposed by inherent limitations of the moving-mesh capabilities and only aim to allow the study on the interactions between multiphysics with an analytic solution.

> This work was published in [!citep](eltawila2025) where more details can be found.

Reviewer: ```suggestion
This work was published in [!cite](ELTAWILA2026112068), where more details can be found.
```

\begin{equation}
  \kappa(T) = \kappa_0 T(x).
  \label{eqn:conductivity}
\end{equation}

The differential length at every point along the slab expands upon heating, where temperature is also varying, in proportion to the local thermal strain $\epsilon_x$, such as

\begin{equation}
  \ell(x) = \ell_0 (\epsilon_x +1),
  \label{eqn:ell_eqn}
\end{equation}

where $\ell(x)$ and $\ell_0$ are differential length elements in the heated and unheated slab, respectively. The thermal strain at any location $x$ is given by

\begin{equation}
  \epsilon_x(T)=\int_{T_0}^T \alpha(T) dT,
  \label{eqn:epsilon}
\end{equation}

where the linear expansion coefficient $\alpha(T)$ is given by an inverse square root dependence on temperature,

\begin{equation}
  \alpha(T)\equiv\frac{1}{2\sqrt{T_0T}}.
  \label{eqn:alpha}
\end{equation}

We use total mass preservation to get material density change from the initial to the final state, starting from:

\begin{equation}
  \rho(x)\ell(x) = \rho_0\ell_0, \label{eqn:mass_conservation}
\end{equation}

to get the relation

\begin{equation}
  \label{eqn:rho_t}
  \rho(x) = \rho_0\sqrt{\frac{T_0}{T(x)}}.
\end{equation}

## Analytic solutions

> The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) and the updates made in [!citep](eltawila2025) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.

Reviewer: ```suggestion
The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) and the updates made in [!cite](ELTAWILA2026112068) for full details on the derivation of these analytical solutions for verification. Here we present the final formulas that we use later for comparison with our models.
```

  \Sigma_t(x) \phi(x)q = \frac{P}{L},
  \label{eqn:heating}
\end{equation}

the neutron flux distribution is given by

\begin{equation}
  \phi(x) = \phi_0 \sqrt{1-\frac{(\lambda-1)P^2x^2}{L^2q^2\phi^2_0}},
  \label{eqn:flux_solution}
\end{equation}

and the temperature distribution is given by

\begin{equation}
  T(x) = \frac{q\Sigma_{t,0}T_0L}{P} \phi(x) ;.
  \label{eqn:temperature}
\end{equation}

Note that [eqn:flux_solution] is identical to the formula for an ellipse rather than the cosine shape normally seen with constant cross sections in finite slab geometries, as a result of the benchmark physics coupling.

The solution for the eigenvalue, $\lambda$, is given by

\begin{equation}
  \lambda = \frac{1}{2} \left(1+\sqrt{1+\frac{16q^2\phi^2_0}{P^2}} \right).
  \label{eqn:eigenvalue}
\end{equation}

Using manufactured parameters for the benchmark specifications (e.g., $L_0$, $P$), the values for the microscopic total cross section $\sigma_{t,0}$ and convective heat transfer coefficient $h$ are determined such that the temperature distribution also satisfies the manufactured scenario where the flux and temperature distributions have the same shape as given by [eqn:temperature].

> The final length of the heated slab was evaluated using the flux and temperature solutions [eqn:flux_solution],[eqn:temperature], which were substituted into the mass conservation equation, [eqn:mass_conservation]. Integrating both sides gives the total length of the heated slab implicitly by preserving the total mass from the initial to the final state resulting

Reviewer: ```suggestion
The final length of the heated slab was evaluated using the flux and temperature solutions [eqn:flux_solution] and [eqn:temperature], which were substituted into the mass conservation equation, [eqn:mass_conservation]. Integrating both sides gives the total length of the heated slab implicitly by preserving the total mass from the initial to the final state, resulting in
```

\begin{equation}
  \rho_0\int_{-L/2}^{L/2}\sqrt{\frac{T_0}{T(x)}}dx=\rho_0L_0.
\end{equation}

The final length of the heated slab $L$ is then given by

\begin{equation}
  L = \sqrt[3]{\Bigg(\frac{L_0}{_2F_1(1/4,1/2,3/2,1/\lambda)}\Bigg)^4\Bigg(\frac{\frac{q^2\phi^2_0}{P(\lambda-1)}-\frac{P}{4}}{\kappa_0T_0^2(1-1/\lambda)}\Bigg)}
  \label{eqn:length},
\end{equation}

where $_2F_1$ is the Gauss hypergeometric function.

> To ensure a physically meaningful solution (i.e., the heat flux leaving the slab is consistent with the $T(\pm L/2)-T_0$), there is an additional constraint on the manufactured temperature distribution

Reviewer: ```suggestion
To ensure a physically meaningful solution (i.e., the heat flux leaving the slab is consistent with the $T(\pm L/2)-T_0$), there is an additional constraint on the manufactured temperature distribution:
```

\begin{equation}
  T\left(\pm\frac{L}{2}\right)\geq T_0.
  \label{eqn:constraint}
\end{equation}

Substituting [eqn:flux_solution] and [eqn:temperature] into [eqn:constraint] gives a minimum condition for power in terms of the other slab parameters, as

\begin{equation}
  P \geq \frac{4\kappa_0T_0^2}{L(\lambda-1)}.
  \label{eqn:power_condition}
\end{equation}

> The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ `cm` with $P = 1.0\times10^{22}$ `eV/s`.

Reviewer: ```suggestion
The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ cm with $P = 1.0\times10^{22}$ eV/s.
```

!table id=tab:params caption=Example set of model parameters [!citep](griesheimer2022analytical).
| Parameter                  | Value                   | Unit                              |
| :- | :- | :- |
| $\rho_0$                   | $1.2$                         | `g.cm`$^{-3}$``                          |
| $L_0$                      | $100$                         | `cm`                                  |
| $A$                        | $180$                         | `g/mol`$^{-1}$``                        |
| $T_0$                      | $293$                         | `K`                                   |
| $q$                        | $1.0 \times 10^{8}$                       | `eV`                                  |
| $\kappa_0$               | $1.25 \times 10^{19}$                    | `eV/s/cm/K`$^{-2}$``      |
| $\phi_0$                  | $2.5\times 10^{14}$                      | `#/cm`$^2$`/s`                  |
| $\nu\Sigma_f/\Sigma_t$     | $1.5$                         | `-`                                   |
| $\Sigma_s/\Sigma_t$        | $0.45$                        | `-`                                   |

> | $P$                        | $1.0\times 10^{22}$                     | `eV/s`                         |

Reviewer: ```suggestion
| $\rho_0$                   | $1.2$                         | g.cm$^{-3}$                          |
| $L_0$                      | $100$                         | cm                                  |
| $A$                        | $180$                         | g/mol$^{-1}$                        |
| $T_0$                      | $293$                         | K                                   |
| $q$                        | $1.0 \times 10^{8}$                       | eV                                  |
| $\kappa_0$               | $1.25 \times 10^{19}$                    | eV/s/cm/K$^{-2}$      |
| $\phi_0$                  | $2.5\times 10^{14}$                      | #/cm$^2$/s                  |
| $\nu\Sigma_f/\Sigma_t$     | $1.5$                         | -                                   |
| $\Sigma_s/\Sigma_t$        | $0.45$                        | -                                   |
| $P$                        | $1.0\times 10^{22}$                     | eV/s                         |
```

# Problem model

!alert! construction title=To run this model you need to build your application with the S$_2$ OpenMC patch
This model is using S$_2$ neutron transport to compare with the 1D Analytic benchmark. To replicate

> the results, you have to build your application with [this](https://github.com/meltawila/openmc_S2transport) patch of OpenMC. you can do this simply with:

Reviewer: ```suggestion
the results, you have to build your application with [this](https://github.com/meltawila/openmc_S2transport) patch of OpenMC. You can do this simply with:
```

  id=fig:schematic
  caption=Data transfers between OpenMC and MOOSE and conceptual depiction of the different meshes used.
  style=display:block;margin-left:auto;margin-right:auto;width:60%;

We show the input files for a coarse mesh here with more details on results convergence shown in the next section.

The OpenMC input files is as follows:

!listing /test/tests/neutronics/expanding_slab/model.py language=python

The neutronics input file is as follows:

!listing /test/tests/neutronics/expanding_slab/openmc.i

The thermomechanics input file is as follows:

!listing /test/tests/neutronics/expanding_slab/solid.i

It is important to note that incremental (hypoelastic) formulations of finite strain behavior inherently introduce inaccuracies arising from the time integration error. This limitation is inherent to this formulation and therefore exists in the MOOSE implementation and the implementations of other major commercial codes. The magnitude of these errors grows with increasing strain, so this typically becomes a practical issue only for very large strains, for which hyperelastic models are more appropriate [!citep](belytschko2003).

> The main sources of error in this coupled model are: (i) the statistical error from use of finite particles in OpenMC; (ii) the user-input cross-section data, which is defined with 1 `K` $\Delta$T spacings used in all cases while taking the nearest temperature point cross section; and (iii) discretization error from a finite spatial mesh in both OpenMC (temperature feedback resolution) and MOOSE thermomechanics. Additional sources of error are present, but anticipated to be negligible due to the use of fine settings for these simulation parameters: (i) finite Picard iterations, and (ii) nonzero nonlinear tolerances in the MOOSE thermomechanics model.

Reviewer: ```suggestion
The main sources of error in this coupled model are: (i) the statistical error from use of finite particles in OpenMC; (ii) the user-input cross-section data, which is defined with 1 K $\Delta$T spacings used in all cases while taking the nearest temperature point cross section; and (iii) discretization error from a finite spatial mesh in both OpenMC (temperature feedback resolution) and MOOSE thermomechanics. Additional sources of error are present, but anticipated to be negligible due to the use of fine settings for these simulation parameters: (i) finite Picard iterations, and (ii) nonzero nonlinear tolerances in the MOOSE thermomechanics model.
```

## Results

> We see the heating result in [fig:heatingsol] compared against the analytic solution, as well as flux and temperature solutions in [fig:fluxsol] and [fig:tempsol]. Figures are only included for $P=1.0 \times 10^{22}$ `eV/s` and on a fine mesh where the number of OpenMC and mesh elements is $N = 100$, since all results have a similar shape. Both the heating and flux tally everywhere agrees with the analytic solution within 3$\sigma$.

Reviewer: ```suggestion
We see the heating result in [fig:heatingsol] compared against the analytic solution, as well as flux and temperature solutions in [fig:fluxsol] and [fig:tempsol]. Figures are only included for $P=1.0 \times 10^{22}$ eV/s and on a fine mesh where the number of OpenMC and mesh elements is $N = 100$, since all results have a similar shape. Both the heating and flux tally everywhere agree with the analytic solution within 3$\sigma$.
```

!media figures/slab_benchmark_heatingsol.png
  id=fig:heatingsol
  caption=Heating result from OpenMC for N = 100 mesh elements compared against the analytic solution.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

!media figures/slab_benchmark_fluxsol.png
  id=fig:fluxsol
  caption=Flux result from OpenMC for N = 100 mesh elements compared against the analytic solution.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

!media figures/slab_benchmark_tempsol.png
  id=fig:tempsol
  caption=Temperature result from MOOSE for N = 100 mesh elements compared against the analytic solution.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

[tab:results] shows results convergence with mesh refinement.

> !table id=tab:results caption=Results and corresponding errors with different mesh sizes [!citep](eltawila2025).

Reviewer: ```suggestion
!table id=tab:results caption=Results and corresponding errors with different mesh sizes [!citep](ELTAWILA2026112068).
```

\end{equation}

where $\phi_a$ is the analytic flux solution, $\phi_{sln.}$ is the Cardinal flux solution, $T_a$ is the analytic temperature solution, and $T_{sln.}$ is the Cardinal temperature solution.

!media figures/slab_benchmark_fluxconv.png
  id=fig:fluxconv
  caption=Convergence of flux relative L$^2$ norm.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

!media figures/slab_benchmark_tempconv.png
  id=fig:tempconv
  caption=Convergence of temperature relative L$^2$ norm.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

A first-order convergence of the temperature is achieved with mesh refinement, while the flux starts to converge following a first-order trend then stagnates where other error components dominate as discussed earlier. The stochastic noise error influence is highlighted by keeping the number of particles constant for each line, where we see further convergence when there is less stochastic noise.

We hypothesize that the observed first-order convergence arises from the use of cell tallies in OpenMC, where heating results are tallied as constant monomials. This may be limiting the convergence rate of the coupled solve.

> More details and discussion can be found in [!citep](eltawila2025)

Reviewer: ```suggestion
More details and discussion can be found in [!cite](ELTAWILA2026112068).
```
